src/bin/fish.rs

```diff
@@ -71,9 +71,99 @@
 
 const DOC_DIR: &str = env!("DOCDIR");
 const DATA_DIR: &str = env!("DATADIR");
+const DATA_DIR_SUBDIR: &str = env!("DATADIR_SUBDIR");
 const SYSCONF_DIR: &str = env!("SYSCONFDIR");
 const BIN_DIR: &str = env!("BINDIR");
 
+#[cfg(feature = "installable")]
+fn install(noconfirm: bool) {
+    use rust_embed::RustEmbed;
+
+    #[derive(RustEmbed)]
+    #[folder = "share/"]
+    struct Asset;
+
+    use std::fs;
+    use std::io::ErrorKind;
+    use std::io::Write;
+    use std::io::{stderr, stdin};
+    let (_, home) = fish::env::get_user_home();
+    let Some(home) = home else {
+        eprintln!("Can't find $HOME",);
+        std::process::exit(1);
+    };
+    let dir = PathBuf::from(home).join(DATA_DIR).join(DATA_DIR_SUBDIR);
+
+    // TODO: Translation,
+    // FLOG?
+    // - Install: Translations
+    // - Install: Manpages (build via build.rs)
+    // - Don't install: __fish_build_paths.fish.in
+    if !noconfirm {
+        eprintln!(
+            "This will write fish's data files to '{}'.\n\
+             Please enter 'yes' to continue.",
+            dir.display()
+        );
+        eprint!("> ");
+        let _ = stderr().flush();
+
+        let mut input = String::new();
+        stdin()
+            .read_line(&mut input)
+            .expect("Failed to read from stdin");
+
+        if input != "yes\n" {
+            eprintln!("Exiting without writing any files");
+            std::process::exit(1);
+        }
+    } else {
+        eprintln!("Installing fish's data files to '{}'.", dir.display());
+    }
+
+    // Remove the install directory first, to clean out any removed files.
+    if let Err(err) = fs::remove_dir_all(dir.clone()) {
+        if err.kind() != ErrorKind::NotFound {
+            eprintln!("Removing '{}' failed: {}", dir.display(), err);
+            std::process::exit(1);
+        }
+    }
+
+    for file in Asset::iter() {
+        let path = dir.join(file.as_ref());
+        let Ok(_) = fs::create_dir_all(path.parent().unwrap()) else {
+            eprintln!(
+                "Creating directory '{}' failed",
+                path.parent().unwrap().display()
+            );
+            std::process::exit(1);
+        };
+        let res = File::create(&path);
+        let Ok(mut f) = res else {
+            eprintln!("Creating file '{}' failed", path.display());
+            continue;
+        };
+        // This should be impossible.
+        let d = Asset::get(&file).expect("File was somehow not included???");
+        f.write_all(&d.data).expect("FAILED TO WRITE");
```

Comment:
nit: should also do a non-panicking error check for `write_all()`,
in case we run out of disk space or something